Configure PHPStan with Symfony & Doctrine support

SergeyGrinenko opened the pull request:
Configuration includes:
- Added PHPStan, phpstan-symfony, phpstan-doctrine to require-dev
- Added doctrine fixtures as dev dependency
- Added phpstan script to composer.json
- Created phpstan.dist.neon with:
  - Level 4 static analysis
  - Symfony containerXmlPath integration
  - Doctrine extension config
  - Project paths (bin, config, public, src, tests)
- Added phpstan.md containing installation and usage instructions
- Warmed up Symfony dev cache to generate container XML

Project now supports reliable static analysis with framework-level type awareness.

SergeyGrinenko committed:
configure PHPStan with Symfony & Doctrine support

ihpr commented on phpstan.md:
```diff
@@ -0,0 +1,37 @@
+# PHPStan - Static Analysis
```
@SergeyGrinenko Move to `docs/` Add a link to `README`. By default, Getting Started is about a quick start, which means it must be set up for dev environment. For prod, there may be a separate installation (Getting started guide) in the docs.

SergeyGrinenko committed:
Add PHPStan dev script, docs and CI workflow

SergeyGrinenko committed:
configure PHP-CS-Fixer for code style checks

SergeyGrinenko commented:
SCRUM-29: centralize documentation and document CI setup

- moved all existing documentation about phpstan and cs-fixer into the /docs directory
- added documentation related to CI configuration and setup

ihpr commented on .github/workflows/php-cs-fixer.yml:
```diff
@@ -0,0 +1,26 @@
+              uses: actions/checkout@v4
+
+            - name: Setup PHP 8.4
+              uses: shivammathur/setup-php@v2
```
@SergeyGrinenko can we trust the author? Better to change the image if we can't trust them?

ihpr commented on .php-cs-fixer.dist.php:
```diff
@@ -0,0 +1,66 @@
+        __DIR__ . '/bin',
+        __DIR__ . '/tools',
+    ])
+    ->exclude([
```
@SergeyGrinenko Do we need this?

ihpr commented on .php-cs-fixer.dist.php:
```diff
@@ -0,0 +1,66 @@
+    ->setUsingCache(true)
+    ->setCacheFile(__DIR__ . '/var/.php-cs-fixer.cache')
+    ->setRules([
+        '@PSR12'   => true,
```
@SergeyGrinenko Review the config again.

SergeyGrinenko committed:
SCRUM-29 Reconfigured CS-Fixer and PHPStan

CS-Fixer:
-Updated .php-cs-fixer.dist.php using the Symfony ruleset remove unnecessary and default rules.

CI:
- Replaced the `shivammathur/setup-php@v2` action with the official `php:8.4-cli` container and manual Composer installation, removing the dependency on an unverified third-party action

SergeyGrinenko committed:
SCRUM-29 fix CI dependencies for PHPStan and CS-Fixer

CI:
- Install git, unzip and zip extension in php:8.4-cli container
- Ensure composer install can download dist or source packages for workflows

CS-Fixer:
-Updated .php-cs-fixer.dist.php using the Symfony ruleset remove unnecessary and default rules.

CI:
- Replaced the `shivammathur/setup-php@v2` action with the official `php:8.4-cli` container and manual Composer installation, removing the dependency on an unverified third-party action

SergeyGrinenko committed:
chore: SCRUM-29 fix CS-Fixer configuration in CI

CS-Fixer:
- Mark the GitHub Actions workspace as a safe Git directory to avoid "dubious ownership" errors during php-cs-fixer runs
- Exclude auto-generated config/reference.php from the finder so it is not reformatted on every run and does not break the cs:check workflow

SergeyGrinenko committed:
chore: SCRUM-29 resolved conflicts and updated configuration PHPStand and CS-Fixer

CS-Fixer:
- Exclude auto-generated config/reference.php from the finder so it is not reformatted on every run and does not break the cs:check workflow

PHPStan:
Ensured required PHPStan packages are installed automatically and Symfony cache warmup when the dev container XML is missing

CPAprince committed:
Merge branch 'develop' into feature/SCRUM-29-Configure-PHPStan
